www-src/app/lib/android_picture_folder_handler.coffee

```diff
@@ -0,0 +1,67 @@
+DesignDocuments = require "../replicator/design_documents"
+log = require("./persistent_log")
+    prefix: "AndroidPictureFolderHandler"
+    date: true
+
+
+module.exports = class AndroidPictureFolderHandler
+
+
+    ERROR_FOLDER_EXIST: "Pictures folder isn't created."
+
+
+    constructor: (@deviceName, @db) ->
+        @deviceName ?= app.replicator.config.deviceName
+        @db ?= app.replicator.db
+
+
+    getOrCreate: (callback) ->
+        log.info "getOrCreate"
+
+        @get (err, folderPictures) =>
+            if err
+                if err.message is @ERROR_FOLDER_EXIST
+                    return @create callback
+
+                return callback err
+
+            callback null, folderPictures
```

Comment:
It can be shortened to:

```coffee
@get (err, folderPictures) =>
    if err?.message is @ERROR_FOLDER_EXIST
        @create callback
    else
        callback err, folderPictures
```